atomic xtensa  build fail

**Describe the bug**
Not all xtensa architectures support the s32c1i and SCOMPARE1 registers
code:
```
#if defined(CONFIG_ATOMIC_OPERATIONS_BUILTIN)
/* Default.  See this file for the Doxygen reference: */
#include <sys/atomic_builtin.h>
#elif defined(CONFIG_ATOMIC_OPERATIONS_ARCH)-----------------------------------------------------------here
/* Some architectures need their own implementation */
# ifdef CONFIG_XTENSA
/* Not all Xtensa toolchains support GCC-style atomic intrinsics */
# include <arch/xtensa/atomic_xtensa.h>
# endif
#else
/* Generic-but-slow implementation based on kernel locking and syscalls */
#include <sys/atomic_c.h>
#endif

static ALWAYS_INLINE
atomic_val_t xtensa_cas(atomic_t *addr, atomic_val_t oldval,
			atomic_val_t newval)
{
	__asm__ volatile("wsr %1, SCOMPARE1; s32c1i %0, %2, 0" ---------------------------------------------here
			 : "+r"(newval), "+r"(oldval) : "r"(addr) : "memory");

	return newval; /* got swapped with the old memory by s32c1i */
}
```
log：
```
atomic_xtensa.h:35: Error: unknown opcode or format name 's32c1i'
atomic_xtensa.h:35: Error: invalid register 'SCOMPARE1' for 'wsr' instruction
atomic_xtensa.h:35: Error: unknown opcode or format name 's32c1i'
xt-xcc.exe ERROR:  C:/usr/xtensa/XtDevTools/install/tools/RI-2020.5-win32/XtensaTools/bin/xt-as.exe returned non-zero status 1
```
At least the code needs to be judged as follows,or distinguish the architecture used
```
if (XCHAL_HAVE_S32C1)
..............
```

The kconfig I saw opens the ATOMIC_OPERATIONS_ARCH configuration by default.

zephyr\arch\Kconfig：
```
config XTENSA
	bool
	select ARCH_IS_SET
	select HAS_DTS
	select USE_SWITCH
	select USE_SWITCH_SUPPORTED
	select ATOMIC_OPERATIONS_ARCH          -----------------------------------------------------------here
	help
	  Xtensa architecture
```

SECOND:
I found another problem when I enabled ATOMIC_OPERATIONS_BUILTIN
```
config SOC_SERIES_MYSOC
	bool "Xtensa hifi4 core"
	select XTENSA
	select XTENSA_HAL
	select ATOMIC_OPERATIONS_BUILTIN 
```
log:
```
In file included from E:/work/zephyr/include/sys/atomic.h:29,
                 from E:/work/zephyr/include/kernel_includes.h:21,
                 from E:/work/zephyr/include/kernel.h:17,
                 from E:/work/zephyr/include/init.h:11,
                 from E:/work/zephyr/include/device.h:29,
                 from E:/work/zephyr/kernel/include/kernel_offsets.h:6,
                 from E:/work/zephyr/arch/xtensa/core/offsets/offsets.c:7:
E:/work/zephyr/include/sys/atomic_builtin.h:304:29: error: syscalls/atomic.h: No such file or directory
```
Under the build directory I only found "ZCAN-sdk\build\zephyr\include\generated\syscalls\atomic_c.h" but no "atomic.h"
CONFIG_ATOMIC_OPERATIONS_C and CONFIG_ATOMIC_OPERATIONS_BUILTIN and CONFIG_ATOMIC_OPERATIONS_ARCH are enabled by me at the same time. but there is no warning that they cannot be enabled at the same time
        
code:
```
#ifdef CONFIG_ATOMIC_OPERATIONS_C
#include <syscalls/atomic.h>
#endif
```


My current change plan can build the code:	del "select ATOMIC_OPERATIONS_ARCH"  & CONFIG_XTENSA_NO_IPC=y & CONFIG_ATOMIC_OPERATIONS_C=y

zephyr\arch\Kconfig：
```
config XTENSA
	bool
	select ARCH_IS_SET
	select HAS_DTS
	select USE_SWITCH
	select USE_SWITCH_SUPPORTED
	help
	  Xtensa architecture
```

--- a/arch/Kconfig
+++ b/arch/Kconfig
@@ -103,7 +103,7 @@ config XTENSA
 	select HAS_DTS
 	select USE_SWITCH
 	select USE_SWITCH_SUPPORTED
-	select ATOMIC_OPERATIONS_ARCH
+	imply ATOMIC_OPERATIONS_ARCH
 	help
 	  Xtensa architecture
 

--- a/include/sys/atomic.h
+++ b/include/sys/atomic.h
@@ -24,9 +24,9 @@ typedef void *atomic_ptr_t;
 
 /* Low-level primitives come in several styles: */
 
-#if defined(CONFIG_ATOMIC_OPERATIONS_BUILTIN)
-/* Default.  See this file for the Doxygen reference: */
-#include <sys/atomic_builtin.h>
+#if defined(CONFIG_ATOMIC_OPERATIONS_C)
+/* Generic-but-slow implementation based on kernel locking and syscalls */
+#include <sys/atomic_c.h>
 #elif defined(CONFIG_ATOMIC_OPERATIONS_ARCH)
 /* Some architectures need their own implementation */
 # ifdef CONFIG_XTENSA
@@ -34,8 +34,8 @@ typedef void *atomic_ptr_t;
 # include <arch/xtensa/atomic_xtensa.h>
 # endif
 #else
-/* Generic-but-slow implementation based on kernel locking and syscalls */
-#include <sys/atomic_c.h>
+/* Default.  See this file for the Doxygen reference: */
+#include <sys/atomic_builtin.h>
 #endif
 
 /* Portable higher-level utilities: */

--- a/include/sys/atomic_builtin.h
+++ b/include/sys/atomic_builtin.h
@@ -300,8 +300,4 @@ static inline atomic_val_t atomic_nand(atomic_t *target, atomic_val_t value)
 }
 #endif
 
-#ifdef CONFIG_ATOMIC_OPERATIONS_C
-#include <syscalls/atomic.h>
-#endif
-
 #endif /* ZEPHYR_INCLUDE_SYS_ATOMIC_BUILTIN_H_ */

